### PR TITLE
Allow for additional arguments to be passed to provider help commands

### PR DESCRIPTION
**Pull request by JacobCallahan**
This change allows users to pass along additional arguments to provider help commands. Providers can then choose to use these arguments as they see fit.

One new possible example is to specify a tower_inventory when listing the available templates for AnsibleTower.

I also added a couple of new logging statements to make it clear what tower inventory is being used by Broker and when.

### CODE REVIEW

File: broker/providers/ansible_tower.py [hunk 1]
```diff
@@ -351,7 +351,7 @@ def _compile_host_info(self, host):
         host_info = {
             "name": host.name,
             "type": host.type,
-            "hostname": host.variables.get("fqdn"),
+            "hostname": host.variables.get("fqdn", host.facts.get("ansible_fqdn")),
```
Comment (ColeHiggins2) [inline broker/providers/ansible_tower.py:354]: this will be very helpful!

**Review by ColeHiggins2 — APPROVED** [review]
Looks good to me !

File: broker/providers/ansible_tower.py (hunk 1 above)
Comment (JacobCallahan) [reply broker/providers/ansible_tower.py:354]: i hope so

Comment (ColeHiggins2) [reply broker/providers/ansible_tower.py:354]: after further investigation. It looks like 'Host' object has no attribute 'facts'.

**Review by Griffin-Sullivan — APPROVED** [review]

**Review by ogajduse — APPROVED** [review]

**Pull request merged**